charts/drax/charts/dashboard/values.yaml

```diff
@@ -12,36 +12,26 @@ bootstrap:
     port: 0
 
 config:
-  config_id: "production"
-  defaultOranNamespace: "default"
-  grafanaURL: "10.20.20.20"
-  grafanaPort: "30300"
-  nodeApiURL: "10.20.20.20"
-  nodePort: "30315"
-  kafkaUrl: "{{ .Release.Name }}-kafka"
-  kafkaPort: "9092"
-  svcOrchestratorHost: "{{ .Release.Name }}-service-orchestrator"
-  svcOrchestratorPort: "80"
-  networkStateMonitorHost: "{{ .Release.Name }}-network-state-monitor"
-  networkStateMonitorPort: "5000"
-  configApiHost: "{{ .Release.Name }}-config-api"
-  configApiPort: "80"
-  svcMonitorHost: "{{ .Release.Name }}-service-monitor"
-  svcMonitorPort: "80"
-  pcixAppPodName: "accelleran-drax-pci-010-pci-xapp-api"
-  pcixAppPort: "80"
-  ksqldbPodName: "10.20.20.20"
-  ksqldbPort: "30533"
-  LteRadioControllerLabel: "4G-Radio-Controller"
-  apiUrl: "10.20.20.20"
-  websocketPort: "31316"
-  topologyExpired: 5000
+  defaultOranNamespace: ""
+  kafkaURL: ""
+  kafkaPort: ""
+  svcOrchestratorHost: ""
+  svcOrchestratorPort: ""
+  networkStateMonitorHost: ""
+  networkStateMonitorPort: ""
+  configApiHost: ""
+  configApiPort: ""
+  svcMonitorHost: ""
+  svcMonitorPort: ""
+  pcixAppPodName: ""
+  pcixAppPort: ""
+  ksqldbPodName: ""
+  ksqldbPort: ""
+  LteRadioControllerLabel: ""
+  kafkaTopics: []
+  topologyExpired: ""
   draxVersionConfigmap: ""
-  kafkaTopics:
-    - "accelleran.drax.5g.ric.raw.cu_state"
-    - "accelleran.drax.5g.ric.raw.ue_measurements"
-    - "accelleran.drax.4g.ric.raw.network_state"
-    - "accelleran.drax.all.ric.processed.anr"
+  grafanaPort: ""
```

Comment:
I'm not sure if we should let charts higher up define really everything? Things like `topologyExpired` could have a sensible default within the dashboard's value so we don't need to put everything within the `drax` values. The only thing I would expect in the `drax` values by default are all the hosts (and ports?) that live within kubernetes (ie services) and then empty values for the same fields in the dashboard values (as there is just no sensible default for a stand-alone dashboard installation given that all it's dependencies are "external").

One exception to the grouping based on hosts is that drax version configmap as it's also something to be injected (so to be handled the same as a host).

I'm also not completely sure about the ports, we should maybe define it in both values files if there is a clear default based on protocol, ...? E.g. for kafka we define port 9092 in both places maybe? I would add that at least as a default to the dasbhoard.

So maybe a proposal of what we should give a default value here:


 ```suggestion
   grafanaPort: 0 # we can use 0 to indicate the type iso an empty string
   kafkaURL: ""
   kafkaPort: 9092
   svcOrchestratorHost: ""
   svcOrchestratorPort: 80
   networkStateMonitorHost: ""
   networkStateMonitorPort: 5000 # I don't find this one clear for I would expect port 80
   configApiHost: ""
   configApiPort: 80
   svcMonitorHost: ""
   svcMonitorPort: 80
   pcixAppPodName: ""
   pcixAppPort: 80
   ksqldbPodName: ""
   ksqldbPort: 80
   draxVersionConfigmap: ""
   defaultOranNamespace: "{{ $.Release.Namespace }}"
   LteRadioControllerLabel: "4G-Radio-Controller"
   kafkaTopics:
     - "accelleran.drax.5g.ric.raw.cu_state"
     - "accelleran.drax.5g.ric.raw.ue_measurements"
     - "accelleran.drax.4g.ric.raw.network_state"
     - "accelleran.drax.all.ric.processed.anr"
   topologyExpired: 5000
```

Reply:
I am still confused about the general approach we want to take,

I feel we want to give the minimum configuration in the drax values level but at the same time, some variables are defined at this level and we remove them so it looks like it goes in contradiction.

For example the variable **defaultOranNamespace** (also the defaultServiceNamespace)

We remove it from drax but the default value is release.namespace so available from drax charts?

The logic way to define it to me would be in drax:
`defaultOranNamespace: "{{ $.Release.Namespace }}"`
but in dashboard:
`defaultOranNamespace: "default"`

Is it correct to take this approach for all variables that are based on drax charts?

Reply:
As for the "contradictions": my rule of thumb that I was trying to get over is everything that has a sensible default is filled in in the dashboard values. So "... give the minimum configuration in the drax values level ..." is worded perfectly in line with that. The "... and we remove them [some variables]..." you're confused about I think is because it's not clear what's happening under the hood with helm itself?

Helm will merge dictionaries, so also our `config` dictionary. This means that all defaults we define within the dashboard values will still be there when used by the drax chart. Unless they get overridden of course, which I would do with for example the hosts. So there is no removal happening. 

So with that in mind, the dashboard values will include for example the topology expiration as the only reason to change it is probably user preference. There are not really other external factors to influence it's value. So we also want the drax chart to take over this value and thus we can omit it over there in the values based on how helm merges the values.
_(For most of these we probably could argue whether these should be set in the helm chart at all with a default value instead of having that default within the application. That's another discussion though.)_

However, an external host (to the dashboard chart) and the drax version configmap reference will both have a "dynamic" value as it depends on how these dependencies are deployed themselves.
In our case that would be with an unpredictable release name that's injected by helm - changing the service name for each install (the service name is resolvable with DNS in the cluster as it's used by us as a "host").
So for the dashboard values we would use the empty values and we indicate to the user to fill it in depending on the specific deployment.

The drax chart is used to install the dashboard as well, so we could see the drax chart as a user of the dashboard chart. At that point the release name and all service names that the dashboard depends on are known (unless something in the values is changed, but then it's on the one installing to bring everything back in sync). Hence we can safely inject their values with the `{{ $.Release.Name }}` as part of the template.

Reply:
For the namespace we should decide what's the most sensible default. Would we want to always use the same namespace as the dashboard or drax  (ie `{{ $.Release.Namespace }}`) or do we always want to default to the `"default"` namespace? We could make an argument for both. Not sure what we should go for to be honest. Having the proposed (different) default namespace for the `dashboard` and `drax` chart also makes perfect sense to me. The only thing we should maybe think about here is consistency between both charts to prevent user confusion as much as possible.

I think the O-RAN namespace is a bit special though to start drawing a generalized conclusion from as it's used to discover/deploy other applications (like the CU-CP) in and not merely some connection detail. So I wouldn't use this approach for all drax values that also occur in the subcharts. The only generalization I can come up with is that we'll have to decide on how to use them in two different contexts: the one of the drax chart and the one in the subchart. So as always: it depends :shrug: